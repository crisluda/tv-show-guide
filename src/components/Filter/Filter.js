import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const Filter = ({ params }) => {
  const getShows = () => {
    const result = [];
    if (params.filter === 'All') {
      params.fetchedShows.map(show => result.push(
        <Link
          key={show.show.id}
          to={`/show/${show.show.id}`}
          show={show.show}
        >
          <Card key={show.show.id} show={show.show} />
        </Link>,
      ));
    } else {
      params.fetchedShows
        .filter(show => show.show.type === params.filter)
        .map(show => result.push(
          <Link
            key={show.show.id}
            to={`/show/${show.show.id}`}
            show={show.show}
          >
            <Card key={show.show.id} show={show.show} />
          </Link>,
        ));
    }
    return result;
  };

  return (
    <div className="row">
      <div className="col s12">
        {getShows()}
      </div>
    </div>
  );
};

Filter.propTypes = {
  fetchedShows: PropTypes.arrayOf(PropTypes.object),
  params: PropTypes.shape({
    fetchedShows: PropTypes.arrayOf(PropTypes.object),
    filter: PropTypes.string,
  }),
};

Filter.defaultProps = {
  fetchedShows: [],
  params: {},
};

export default Filter;
