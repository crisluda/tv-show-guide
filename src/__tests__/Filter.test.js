import React from 'react';
import { render } from '@testing-library/react';
import Filter from '../components/Filter/Filter';

it('renders the Filter component correctly', () => {
  const params = {
    fetchedShows: [],
    filter: 'All',
  };
  const { asFragment } = render(<Filter params={params} />);
  expect(asFragment()).toMatchSnapshot();
});
