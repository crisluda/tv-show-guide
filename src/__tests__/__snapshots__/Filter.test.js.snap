// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Filter component correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col s12"
    />
  </div>
</DocumentFragment>
`;
